### PR TITLE
add v4 loadavg sampler and test

### PR DESCRIPTION
**Pull request by baallan**
porting loadavg sampler to v4 from v3 before I forget.
Is bundled with vmstat in configuration options.

### CODE REVIEW

**Comment by baallan** [comment]
@tom95858 updated build files to split loadavg from vmstat configs.

**Comment by tom95858** [comment]
Hi @baallan, 

Am I misreading this or is it possible for the user to select a combination the 6 metrics in the loadavg file. Doesn't this generate potentially 64 different schema? Is there a reason for this additional complexity? It would seem to make analysis particularly challenging.

**Comment by baallan** [comment]
The plugin (unlike meminfo, vmstat, and procstat) automatically (no admin name invention required) creates a schema name uniquely matching the schema content. The default (all metrics in the source file) is simply loadavg. Every other combination of metric  requests has a unique and documented schema name that will be identical for every other administrator/system in the world that requests the same schema content. 

What could be simpler, other than moving on to schema-free user-oriented designs?

File: ldms/src/sampler/loadavg/loadavg.c [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+}
+
+/**
+ * \brief Configuration
+ *
+ * config name=loadavg producer_name=<name> instance_name=<instance_name> [component_id=<compid> schema=<sname>] [metrics=list]
+ *     producer_name    The producer id value.
+ *     instance_name    The set name.
+ *     component_id     The component id. Defaults to zero
+ *     sname            Optional schema name. Defaults to meminfo
```
Comment (tom95858) [inline ldms/src/sampler/loadavg/loadavg.c:274]: Could you please fix this copy and paste error? I think this is supposed to say 'loadavg'.

File: ldms/src/sampler/loadavg/loadavg.c [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+		goto out;
+	}
+
+#define LINE_FMT "%lg %lg %lg %" SCNu64 "/%" SCNu64 " %" SCNu64 "\n"
+	for (i = 0; i < METLEN; i++) {
+		metname[0].v.u = 0;
+	}
+	rc = sscanf(lbuf, LINE_FMT, &metname[0].v.d, &metname[1].v.d, &metname[2].v.d,
+		&metname[3].v.u, &metname[4].v.u, &metname[5].v.u);
+	/*
```
Comment (tom95858) [inline ldms/src/sampler/loadavg/loadavg.c:369]: Did you mean to leave this here? Maybe ifdef it?

File: ldms/src/sampler/loadavg/loadavg.c [hunk 3]
```diff
@@ -0,0 +1,438 @@
+/* -*- c-basic-offset: 8 -*-
+ * Copyright (c) 2020 National Technology & Engineering Solutions
+ * of Sandia, LLC (NTESS). Under the terms of Contract DE-NA0003525 with
```
[the remark this answers is not in the document]
Comment (baallan) [reply ldms/src/sampler/loadavg/loadavg.c:3]: @valleydlr @oceandlr I thought I received direction a while back that all new files would be under ntess copyright only, regardless of subcontract origin. Is that incorrect?

[the remark this answers is not in the document]
Comment (tom95858) [reply ldms/src/sampler/loadavg/loadavg.c:3]: In this case, It's not about the subcontract, it's about the copyright in the copied source code. This is not a "new" file, it is a derivative work (of meminfo in this case) and therefore must maintain the copyright of the original work.

If it is a newly developed original work and it is under the subcontract with OGC (i.e. OGC developed it), then we share copyright.  If it is a new file and NTESS developed it, then it has the NTESS copyright only. Similarly for LANL, LBNL, etc...

[the remark this answers is not in the document]
Comment (baallan) [reply ldms/src/sampler/loadavg/loadavg.c:3]: Tom's is not the interpretation I received during the v4 copyright assertion process.
But Ann/Jim will need to clarify as I am not  privy to the contract language either for ntess to doe or for ogc to ntess.

**Comment by tom95858** [comment]
Hi @baallan, thank you for making the changes. I think it is ready to merge modulo the copyright.

File: ldms/src/sampler/loadavg/loadavg.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (tom95858) [reply ldms/src/sampler/loadavg/loadavg.c:3]: Hi @ballan, mine is not an "interpretation", your rights are stated in the copyright of the file you copied....
 *
 *      Modified source versions must be plainly marked as such, and
 *      **must not** be misrepresented as being the original software.
 *
A copyright is meaningless if you can simply copy the file and change the copyright statement. Thus the name...a copyright is a statement of your right to copy the property and under what terms. In this case, you can't copy the file and claim that it is your original work.

[the remark this answers is not in the document]
Comment (baallan) [reply ldms/src/sampler/loadavg/loadavg.c:3]: I received clarifications specific to this file from @valleydlr and updated the header as @tom95858 requests. Please review the complete header on loadavg.c.

**Pull request merged**